wolvic/browser/vr/wvr_manager.cc

```diff
@@ -850,4 +875,62 @@ void WvrManager::UpdateLayerBounds(int16_t frame_index,
   CreateOrResizeWebXrSurface(source_size);
 }
 
+device::mojom::XREnvironmentBlendMode
+WvrManager::PickEnvironmentBlendModeForSession(
+    device::mojom::XRSessionMode session_mode) {
+  DCHECK(wvr_api_->get_system_state().enumerationCompleted);
+
+  const auto& display_state = wvr_api_->get_system_state().displayState;
+  auto supportsBlendMode =
+      [&display_state](const mozilla::gfx::VRDisplayBlendMode blend_mode) {
+        for (auto& supported_blend_mode : display_state.blendModes) {
+          if (supported_blend_mode ==
+              mozilla::gfx::VRDisplayBlendMode::_empty) {
+            return false;
+          }
+          if (supported_blend_mode == blend_mode) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+  switch (session_mode) {
+    case device::mojom::XRSessionMode::kImmersiveVr:
+      if (supportsBlendMode(mozilla::gfx::VRDisplayBlendMode::Opaque)) {
+        return device::mojom::XREnvironmentBlendMode::kOpaque;
+      }
+      break;
+    case device::mojom::XRSessionMode::kImmersiveAr:
+      // Prefer Alpha Blend when both Alpha Blend and Additive modes are
+      // supported. This only concerns video see through devices with an
+      // Additive compatibility mode
+      if (supportsBlendMode(mozilla::gfx::VRDisplayBlendMode::AlphaBlend)) {
+        return device::mojom::XREnvironmentBlendMode::kAlphaBlend;
+      } else if (supportsBlendMode(
+                     mozilla::gfx::VRDisplayBlendMode::Additive)) {
+        return device::mojom::XREnvironmentBlendMode::kAdditive;
+      }
+      break;
+    case device::mojom::XRSessionMode::kInline:
+      NOTREACHED();
```

Comment:
NOTREACHED disappears in production builds; it'd preferable to use NOTREACHED_NORETURN 

https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/checks.md#invariant_verification-mechanisms 

Also, why Inline is not valid here ?

Reply:
I'm using `NOTREACHED` on purpouse so that the code falls back to returning the first blend mode in the list